x/twap/genesis.go

```diff
@@ -0,0 +1,34 @@
+package twap
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v11/x/twap/types"
+)
+
+// InitGenesis initializes the capability module's state from a provided genesis
+// state.
+func (k Keeper) InitGenesis(ctx sdk.Context, genState *types.GenesisState) {
+	if err := genState.Validate(); err != nil {
+		panic(err)
+	}
+
+	k.SetParams(ctx, genState.Params)
+
+	for _, twap := range genState.Twaps {
+		k.storeNewRecord(ctx, twap)
```

Comment:
This requires the TWAPs to be sorted by time (or really for the last TWAP per pool to be called last)

We should either do the sorting here, or enforce the relevant property in validate

Reply:
Fixed